content/en/blog/_posts/2023-12-05-image-filesystem.md

```diff
@@ -0,0 +1,171 @@
+This allows for more configurability to Kubernetes and provides support for adding a larger disk for the container storage while keeping the default filesystem untouched.  
+
+One area we need to explain is where/what is Kubernetes writing to disk.
+
+Kubernetes has persistent data and ephemeral data.  The root filesystem of Kubernetes is configurable but it is usually assumed to be `/var/lib/kubelet`.  In the Kubernetes docs, this is sometimes referred to as the root filesystem or the node filesystem.  The bulk of this data can be categorized into ephemeral storage, logs and the container runtime.
+
+**Ephemeral storage**
+
+Pods and containers can require temporary or transient local storage for their operation. The lifetime of this ephemeral storage does not extend beyond the life of the individual pod, and this ephemeral storage cannot be shared across pods.
+
+**Logs**
+
+Kubernetes stores the logs of each container running in `/var/logs`.  These logs are ephemeral and are monitored by Kubelet to make sure that they do not grow too large while the pods are running.
+
+**Container runtime**
+
+The container runtime has two different areas of storage for containers and images. Images are usually denoted as the read-only layer as they are not modified when containers are running. There is a thin layer on top of containers that provides ephemeral storage for containers if the container is writing to the filesystem. This is refered to as the writable layer.
+
+## Container Runtime Configurations
+
+### CRI-O
+
+CRI-O uses a storage configuration file to control how the container runtime stores persistent data and temporary data.
+CRI-O utilizes the [storage library](https://github.com/containers/storage).  
+Some Linux distributions have a manual entry for storage (`man 5 containers-storage.conf`).
+The main configuration for storage is located in `/etc/containers/storage.conf` and one can control the location for temporary data and the root directory.  
+The root directory is where CRI-O stores the persistent data.
+
+```toml
+[storage]
+
+# Default Storage Driver
+driver = "overlay"
+# Temporary storage location
+runroot = "/var/run/containers/storage"
+# Primary Read/Write location of container storage 
+graphroot = "/var/lib/containers/storage"
+```
+
+- Graphroot
+  - Persistent data stored from the container runtime
+  - If SELinux is enabled, this must match the `/var/lib/containers/storage`
+- Runroot
+  - Temporary read/write access for container.
+  - Recommended to have this on a temporary filesystem.
+
+A quick way to relabel your graphroot directory to match `/var/lib/containers/storage`.
+
+```bash
+semanage fcontext -a -e /var/lib/containers/storage /YOUR_STORAGE_PATH
+restorecon -R -v /YOUR_STORAGE_PATH
+```
+
+### Containerd
+
+Containerd uses a .toml configuration file to control where persistent and ephemeral data is stored.
```

Comment:
```suggestion
containerd uses a .toml configuration file to control where persistent and ephemeral data is stored.
```

Reply:
Not only for consistency, but also because it's styled with all lower-case in the official documentation.